storage/perfschema/pfs_instr.cc

```diff
@@ -547,6 +547,19 @@ PFS_thread *create_thread(PFS_thread_class *klass,
     pfs->m_statements_history_index = 0;
     pfs->m_transactions_history_full = false;
     pfs->m_transactions_history_index = 0;
+    pfs->set_enabled(true);
+    pfs->m_disable_instrumentation = false;
+    pfs->set_history(true);
```

Comment:
@dutow Not sure I understand why we have to reassign all these members again

Reply:
Fixed.